Description: CVE-2026-107778 - Avoid null dereference in krb5_rd_cred()
 A malformed KRB-CRED message may have fewer entries in the encrypted
 ticket-info list than it has in the tickets list.  In each loop
 iteration of make_cred_list(), check whether the end of the ticket-info
 list was unexpectedly reached to avoid a null pointer dereference.
Origin: upstream, https://github.com/krb5/krb5/commit/48afa9abb89ab2176bb20624d87d010b9984fc08
Author: TristanInSec <tristan.mtn@gmail.com>
Last-Update: 2026-10-11
---
--- a/src/lib/krb5/krb/rd_cred.c
+++ b/src/lib/krb5/krb/rd_cred.c
@@ -98,6 +98,11 @@ make_cred_list(krb5_context context, krb5_cred *krbcred,
             goto cleanup;
 
         info = encpart->ticket_info[i];
+        if (info == NULL) {
+            /* We unexpectedly reached the end of the encrypted ticket info. */
+            ret = KRB5KRB_AP_ERR_MODIFIED;
+            goto cleanup;
+        }
         ret = krb5_copy_principal(context, info->client, &list[i]->client);
         if (ret)
             goto cleanup;
Description: CVE-2026-11850 - Prevent read overrun in libkdb_ldap berval2tl_data
 In berval2tl_data(), reject inputs of length less than 2 to prevent an
 integer underflow and subsequent read overrun. When bv_len is 0 or 1,
 the unsigned subtraction (bv_len - 2) wraps to a large value which is
 then truncated to uint16_t, causing a subsequent malloc to succeed
 while memcpy reads up to 65534 bytes from a 0-1 byte buffer - a heap
 out-of-bounds read. (The security impact is negligible as the attacker
 would have to control the KDB LDAP server.)
Origin: upstream, https://github.com/krb5/krb5/commit/2a5fd83d4436583f2ddc0e193269a4d800ee45c4
Author: Sebastián Alba <sebasjosue84@gmail.com>
Last-Update: 2026-07-14

--- a/src/plugins/kdb/ldap/libkdb_ldap/ldap_principal2.c
+++ b/src/plugins/kdb/ldap/libkdb_ldap/ldap_principal2.c
@@ -80,6 +80,9 @@ getstringtime(krb5_timestamp);
 krb5_error_code
 berval2tl_data(struct berval *in, krb5_tl_data **out)
 {
+    if (in->bv_len < 2)
+        return EINVAL;
+
     *out = (krb5_tl_data *) malloc (sizeof (krb5_tl_data));
     if (*out == NULL)
         return ENOMEM;
